vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/dataview/ItemCountUnknownComboBoxIT.java

```diff
@@ -64,43 +65,37 @@ public void undefinedSize_switchesToDefinedSize_sizeChanges() {
         verifyItemsSize(getDefaultInitialItemCount());
         verifyFetchForUndefinedSizeCallback(0, Range.withLength(0, pageSize));
 
-        doScroll(120, 400, 1, 50, 200);
+        doScroll(120, 400, 1, 50, 200, "Callback Item 120");
 
-        doScroll(299, actualSize, 2, 150, actualSize);
+        doScroll(299, actualSize, 2, 150, actualSize, "Callback Item 299");
 
         Assert.assertEquals(300, getItems(comboBoxElement).size());
 
         // change callback backend size limit
         setUnknownCountBackendSize(DEFAULT_DATA_PROVIDER_SIZE);
-        // TODO is that relevant to ComboBox?
-        // grid has scrolled to end -> switch to defined size callback
+        // combo box has scrolled to end -> switch to defined size callback
         // -> new size updated and more items fetched
         setCountCallback();
 
-        verifyItemsSize(DEFAULT_DATA_PROVIDER_SIZE);
-        // new rows are added to end due to size increase
-        Assert.assertEquals(301, getItems(comboBoxElement).size());
+        // Open dropdown and scroll to the end again, check the DP size
+        doScroll(299, DEFAULT_DATA_PROVIDER_SIZE, 2, 150, actualSize,
+                "Callback Item 299");
 
-        scrollToItem(comboBoxElement, 500);
-
-        int expectedLastItem = 517;
-        Assert.assertEquals(
-                "ComboBox should be able to scroll after changing to defined " +
-                        "size",
-                expectedLastItem, getItems(comboBoxElement).size());
+        // Check that combo box is scrolled over 'actualSize' after switching
+        // to defined size
+        doScroll(500, DEFAULT_DATA_PROVIDER_SIZE, 2, 150, actualSize,
+                "Callback Item 500");
 
         // switching back to undefined size, nothing changes
         setUnknownCount();
 
-        verifyItemsSize(DEFAULT_DATA_PROVIDER_SIZE);
-        Assert.assertEquals(expectedLastItem, getItems(comboBoxElement).size());
+        doScroll(500, DEFAULT_DATA_PROVIDER_SIZE, 2, 150, actualSize,
+                "Callback Item 500");
 
         // increase backend size and scroll to current end
         setUnknownCountBackendSize(2000);
         // size has been increased again by default size
-        doScroll(1000, 1200, 6, 950, 1100);
-
-        Assert.assertEquals(1001, getItems(comboBoxElement).size());
+        doScroll(1000, 1200, 6, 950, 1100, "Callback Item 1000");
     }
 
     // @Test TODO
```

Comment:
Still TODOs present

Reply:
Needs a clarification what steps to add in these tests, let's discuss it when appropriate.

Reply:
Done.